### PR TITLE
fixing tslint configuration

### PR DESCRIPTION
**Pull request by robertohuertasm**
Fixes 'tslint.json' file and a few `double quotes`. Several minor package updates.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -216,7 +216,7 @@
         "precompile": "rimraf ./out",
         "compile": "tsc",
         "compile:w": "npm run compile -- -w",
-        "lint": "tslint --project tsconfig.json",
+        "lint": "tslint --project tsconfig.json --type-check true",
```
Comment (JimiC) [inline package.json]: I believe `true`  is not necessary

File: tslint.json [hunk 2]
```diff
@@ -13,6 +13,7 @@
       "allow-pascal-case",
       "allow-leading-underscore"
     ],
+    "no-submodule-imports": false,
```
Comment (JimiC) [inline tslint.json]: If we are going to use the setting then better set it to true and exclude `vscode`. 
i. e. `no-submodule-imports: [true, "vscode"]`

Comment (robertohuertasm) [reply tslint.json]: I'm used to have this enabled from my work with `Angular` and `rxjs` but given that here we don't use `webpack` and we don't do any kind of `tree-shaking` I can live with it 😉

Comment (JimiC) [reply tslint.json]: Fine.

**Comment by robertohuertasm** [comment]
Wait to approve it as I'm also updating the Changelog.

**Comment by robertohuertasm** [comment]
Note that I updated the changelog with September 17 as the new release date but we can advance it if you are ok with it.

File: package.json (hunk 1 above)
Comment (robertohuertasm) [reply package.json]: Not really, you're right.

**Pull request merged**